### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by Domon** [brookhong/Surfingkeys#78]
Hi,

Thanks for this great extension. I really love it.

However, it uses so much memory.

For example, when I have 34 tabs opened, it uses 861MB of memory.

![chrome-extension-surfingkeys-annotated](https://cloud.githubusercontent.com/assets/69308/17762191/ce6689fe-6540-11e6-8d9b-0f7c6cb25deb.jpg)

I am just wondering if there is any plan to improve its memory usage? 😄

**Comment by brookhong** [comment]
Thanks for the finding.

I run a simple profiling, and found that two libraries used in Surfingkeys -- ACE and jQuery are top 2 memory consumers.

ACE is to provide vim editor, which is very helpful on input/textarea/select. I'll see if there is a chance is simplify it for Surfingkeys usage.

jQuery is introduced for clean coding, it can be removed by creating some simple replacement.

**Comment by brookhong** [comment]
After some improvement, the memory usage has been largely decreased.

Tested on Version 52.0.2743.116 (64-bit) under Mac OS x 10.11.6.
29 tabs opened in a chrome window.

Before the improvement,
138M
after the improvement,
63M.


The improvement includes
* lazy load of ace.js for vim editor
* refine usages of settings, cache only simple settings in pages, and fetch large settings data on demand.

**Comment by brookhong** [comment]
Please test 0.5.7, and kindly show how it behaves on your PC.

**Comment by Domon** [comment]
I tried 0.5.7. The memory usage has been impressively improved!

I opened around 20 - 40 tabs. The memory usage was between 60 - 300MB.

The following is a screenshot when the extension's memory usage was lower.

![chrome-extension-surfingkeys-memory-usage-improved-annotated](https://cloud.githubusercontent.com/assets/69308/17928208/4398a424-6a2c-11e6-8636-7016ba5cfb60.jpg)

I am going to close this issue. Thanks @brookhong! 😃
